scripts/release/Jenkinsfile

```diff
@@ -0,0 +1,104 @@
+                            ]
+                        ])
+
+                        sh "bash scripts/release/socket.sh \$(cat scripts/release/tmp/instance) ${PASSPHRASE}"
+                }
+                */
+
+                sh 'ssh -i BuilderInstanceKey.pem -A ubuntu@$(cat scripts/release/tmp/instance) bash go/src/github.com/algorand/go-algorand/scripts/release/master-controller/build.sh'
+            }
+        }
+
+        stage("package") {
+            steps {
+                input "GPG remote socket"
+                sh 'ssh -i BuilderInstanceKey.pem -A ubuntu@$(cat scripts/release/tmp/instance) bash go/src/github.com/algorand/go-algorand/scripts/release/master-controller/package.sh'
+            }
+        }
+
+        stage("sign") {
+            steps {
+                sh 'ssh -i BuilderInstanceKey.pem -A ubuntu@$(cat scripts/release/tmp/instance) bash go/src/github.com/algorand/go-algorand/scripts/release/master-controller/sign.sh'
+            }
+        }
+
+        stage("upload") {
+            steps {
+                script {
+                    RSTAMP = sh(returnStdout: true, script: 'scripts/reverse_hex_timestamp')
+                    BRANCH = sh(returnStdout: true, script: 'bash scripts/compute_branch.sh')
+                    CHANNEL = sh(returnStdout: true, script: 'bash scripts/compute_branch_channel.sh ${BRANCH}')
+                    FULLVERSION = sh(returnStdout: true, script: 'bash scripts/compute_build_number.sh -f')
+
+                    // Bash scripts return vars with a trailing [:space:]
+                    BRANCH = BRANCH.replaceAll("\\s","")
+                    CHANNEL = CHANNEL.replaceAll("\\s","")
+                    FULLVERSION = FULLVERSION.replaceAll("\\s","")
+
+                    sh "rm -rf node_pkg/* && mkdir -p node_pkg/${RSTAMP}"
+                    sh "scp -i BuilderInstanceKey.pem -o StrictHostKeyChecking=no -r ubuntu@\$(cat scripts/release/tmp/instance):~/node_pkg/* node_pkg/${RSTAMP}/"
+                    sh "aws s3 sync --exclude dev* --exclude master* --exclude nightly* --exclude stable* --acl public-read node_pkg/${RSTAMP} s3://algorand-dev-deb-repo/releases/${CHANNEL}/${RSTAMP}_${FULLVERSION}/"
+                    // Create the buildlog file.
+                    sh 'ssh -i BuilderInstanceKey.pem -A ubuntu@$(cat scripts/release/tmp/instance) bash go/src/github.com/algorand/go-algorand/scripts/release/upload.sh'
+                    // sh "aws s3 cp --quiet ubuntu@\$(cat scripts/release/tmp/instance)/build_status_${CHANNEL}_${FULLVERSION}.asc.gz s3://algorand-devops-misc/buildlog/${RSTAMP}/"
+                    sh "scp -i BuilderInstanceKey.pem -o StrictHostKeyChecking=no ubuntu@\$(cat scripts/release/tmp/instance):~/build_status_${CHANNEL}_*.asc.gz node_pkg/"
+                    sh "aws s3 cp --quiet node_pkg/build_status_${CHANNEL}_*.asc.gz s3://algorand-devops-misc/buildlog/${RSTAMP}/"
+                }
+            }
+        }
+
+        /*
+        stage("tag") {
```

Comment:
If you want to have a "push button" tag, you can use an "input":
https://jenkins.io/doc/pipeline/steps/pipeline-input-step/

I think that we probably want to reverse this though, and only run the job for tagged commits.

Reply:
That make sense.  The workflow can be manually create a tag, then input that to Jenkins to pass to the build agent.

Yes, I'm already using the input step for both pausing the build and getting user input.